tensorflow/compiler/jit/kernels/xla_ops.cc

```diff
@@ -648,6 +659,12 @@ REGISTER_KERNEL_BUILDER(Name("_XlaCompile")
                         XlaCompileOp);
 
 REGISTER_KERNEL_BUILDER(Name("_XlaRun").Device(DEVICE_CPU), XlaRunOp);
-REGISTER_KERNEL_BUILDER(Name("_XlaRun").Device(DEVICE_GPU), XlaRunOp);
+REGISTER_KERNEL_BUILDER(Name("_XlaRun")
```

Comment:
Wow!  Was this working because TF puts all strings on the host?

Reply:
That would be my guess, but did not investigate
Just noticed it will making this change, and fixed it.